libs/datamodel/core/tests/types/negative.rs

```diff
@@ -256,6 +231,95 @@ fn should_fail_on_native_type_with_invalid_arguments() {
+            unsupportes2    Unsupported("Some random (2,5) do something") 
+        }
+    "#;
+
+    let error = parse_error(dml);
+
+    error.assert_are(&[
+        DatamodelError::new_validation_error(
+        "The type `Unsupported(\"Decimal(10,2)\")` you specified in the type definition for the field `decimal` is supported as a native type by Prisma. Please use the native type notation `Decimal @pg.Decimal(10,2)` for full support.",
+        ast::Span::new(188, 216),
+    ),
+        DatamodelError::new_validation_error(
+            "The type `Unsupported(\"Text\")` you specified in the type definition for the field `text` is supported as a native type by Prisma. Please use the native type notation `String @pg.Text` for full support.",
+            ast::Span::new(246, 265),
+        )
+    ]);
+}
+
+#[test]
+fn should_fail_on_native_type_in_unsupported_mysql() {
+    let dml = r#"
+        datasource pg {
+          provider = "mysql"
+          url = "mysql://"
+        }
+
+        model Blog {
+            id          Int    @id
+            text        Unsupported("Text") 
+            decimal     Unsupported("Float") 
+        }
+    "#;
+
+    let error = parse_error(dml);
+
+    error.assert_are(&[
+        DatamodelError::new_validation_error(
+            "The type `Unsupported(\"Text\")` you specified in the type definition for the field `text` is supported as a native type by Prisma. Please use the native type notation `String @pg.Text` for full support.",
+            ast::Span::new(172, 191),
+        ),
+        DatamodelError::new_validation_error(
+            "The type `Unsupported(\"Float\")` you specified in the type definition for the field `decimal` is supported as a native type by Prisma. Please use the native type notation `Float @pg.Float` for full support.",
+            ast::Span::new(217, 237),
+        )
+    ]);
+}
+
+#[test]
+fn should_fail_on_native_type_in_unsupported_sqlserver() {
+    let dml = r#"
+        datasource pg {
+          provider = "sqlserver"
+          url = "sqlserver://"
+        }
+
+        model Blog {
+            id          Int    @id
+            text        Unsupported("Text") 
+            decimal     Unsupported("Real") 
+        }
+    "#;
+
+    let error = parse_error(dml);
+
+    error.assert_are(&[
+        DatamodelError::new_validation_error(
+            "The type `Unsupported(\"Text\")` you specified in the type definition for the field `text` is supported as a native type by Prisma. Please use the native type notation `String @pg.Text` for full support.",
+            ast::Span::new(180, 199),
+        ),
+        DatamodelError::new_validation_error(
+            "The type `Unsupported(\"Real\")` you specified in the type definition for the field `decimal` is supported as a native type by Prisma. Please use the native type notation `Float @pg.Real` for full support.",
+            ast::Span::new(225, 244),
+        )
+    ]);
+}
```

Comment:
No, this currently does only work if it exactly matches our casing. This is because it uses the existing parser unmodified currently. We could switch to a custom parser for this purpose which would increase our hit rate for helpful hints, but probably also get us some falsely positive validation errors. Since there is nothing intuitive a user could do to avoid the validation errors I decided to be careful for now.

Reply:
I added a test case showing that it does not work though.